### DISCUSSION TITLE
treeedb: Generate types, relations, and facts that represent ASTs of 7+ languages

### DISCUSSION THREAD

**Discussion by langston-barrett** [souffle-lang/souffle#2345] [Show and tell]
[`treeedb`](https://github.com/langston-barrett/treeedb) makes it easier to start writing a source-level program analysis in Soufflé. First, `treeedb` generates Soufflé types and relations that represent a program's AST. Then, `treeedb` parses source code and emits facts that populate those relations.

`treeedb` currently supports analysis of these languages:

- C
- C#
- Java
- JavaScript
- Rust
- Soufflé
- Swift

`treeedb`'s parsers and ASTs are based on tree-sitter grammars, and it's very easy to add support for any language with a tree-sitter grammar.

Please see the repo for more information! https://github.com/langston-barrett/treeedb

**Comment by b-scholz** [comment]
I have created a fork under souffle-lang for visibility.
